src/components/letter/representative-apis.tsx

```diff
@@ -0,0 +1,275 @@
+            address_zip: address.zip,
+            address_country: "US",
+          },
+          officeName: office.name,
+          district,
+          levels: office.levels,
+          roles: office.roles,
+          link: official.urls?.[0],
+        };
+      })
+      .filter((a) => a !== undefined) as OfficialAddress[];
+  });
+};
+
+export const fetchRepsFromGoogle = async ({
+  address,
+  googleApiKey,
+  restricts,
+}: {
+  address: string;
+  googleApiKey: string;
+  restricts?: OfficialRestrict[];
+}): Promise<OfficialAddress[]> => {
+  const params = new URLSearchParams({
+    address,
+    key: googleApiKey,
+  }).toString();
+
+  const res = await fetch(
+    "https://www.googleapis.com/civicinfo/v2/representatives?" + params
+  );
+  const json = await res.json();
+
+  return mungeReps(json as GoogleCivicRepsResponse, restricts);
+};
+
+export const fetchRepsFromBlackmad = async ({
+  address,
+  restricts,
+}: {
+  address: string;
+  restricts?: OfficialRestrict[];
+}): Promise<OfficialAddress[]> => {
+  const latLng = await getGeocode({ address }).then((results) =>
+    getLatLng(results[0])
+  );
+
+  const { lat, lng } = latLng;
+
+  const res = await fetch(
+    `https://city-council-api.herokuapp.com/lookup?lat=${lat}&lng=${lng}`
+  );
+
+  const json = (await res.json()) as BlackmadCityCountilResponse;
+
+  // const districtEntry = data.data.find((o) => Boolean(o.district));
+  // if (districtEntry) {
+  //   updateField("YOUR DISTRICT", districtEntry.district);
+  // }
+  return mungeCityCouncil(json, restricts);
+};
+
+/** Is this offical a city council person
+ *
+ * @param {LevelsAndRoles} official offical to check
+ * @return {boolean} is this offical a city council person
+ */
+const isCityCouncilOffical = (official: LevelsAndRoles) => {
+  return (
+    official.roles?.includes("legislatorUpperBody") &&
+    official.levels?.includes("locality")
+  );
+};
+
+export const fetchReps = async ({
+  address,
+  googleApiKey,
+  setIsSearching,
+  restricts,
+}: {
+  address: string;
+  googleApiKey: string;
+  setIsSearching: (isSearching: boolean) => void;
+  restricts?: OfficialRestrict[];
+}): Promise<OfficialAddress[]> => {
+  setIsSearching(true);
+
+  const cityCouncilOnly =
+    restricts?.length === 1 &&
+    restricts?.[0]?.level === "locality" &&
+    restricts?.[0]?.role === "legislatorUpperBody";
+
+  // Make two calls
+  // One to google, which has perfect information as far as I can tell for mayorial level and up everywhere
+  // Second to blackmad's city council API which tries to fill holes in major cities where google's city council coverage is lacking (SF, NYC)
+  let googlePromise: Promise<OfficialAddress[]> = Promise.resolve([]);
+  if (!cityCouncilOnly) {
+    googlePromise = fetchRepsFromGoogle({ address, googleApiKey, restricts });
+  }
+
+  const blackmadPromise = fetchRepsFromBlackmad({ address, restricts });
+
+  const googleResponse = await googlePromise;
```

Comment:
`Promise.all` these?

Reply:
they're being used individually in the following logic - what would a Promise.all help with here?

const [googleResponse, blackmadResponse] = await Promise.all([googlePromise, blackmadPromise])? ... this is less good to me because you would run the risk of getting the inputs out of order